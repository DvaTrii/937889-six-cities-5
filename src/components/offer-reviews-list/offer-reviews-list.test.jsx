import React from "react";
import renderer from "react-test-renderer";
import OfferReviewsList from "./offer-reviews-list.tsx";

it(`Should OfferReviewsList render properly`, () => {
  const reviews = [
    {
      id: 300,
      date: 1605357744252,
      userInfo: {
        avatar: `img/avatar-max.jpg`,
        name: `Tom`
      },
      text: `Lorem ipsum dolor sit et dolore magna aliqua. Ut enim ad minim veniam`,
      rating: 4
    },
    {
      id: 333,
      date: 1605357744253,
      userInfo: {
        avatar: `img/avatar-max.jpg`,
        name: `Bob`
      },
      text: `Lorem ipsum dolor sir incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`,
      rating: 3
    },
  ];

  const tree = renderer
    .create(
        <OfferReviewsList
          reviews={reviews}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
