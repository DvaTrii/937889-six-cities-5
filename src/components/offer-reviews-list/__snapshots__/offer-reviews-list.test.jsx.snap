// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferReviewsList render properly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__review"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Tom
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Lorem ipsum dolor sit et dolore magna aliqua. Ut enim ad minim veniam
      </p>
      <time
        className="reviews__time"
        dateTime="April 2019"
      >
        April 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__review"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Bob
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Lorem ipsum dolor sir incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
      </p>
      <time
        className="reviews__time"
        dateTime="undefined"
      />
    </div>
  </li>
  <li
    className="reviews__review"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Jess
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Lorem eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
      </p>
      <time
        className="reviews__time"
        dateTime="undefined"
      />
    </div>
  </li>
  <li
    className="reviews__review"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Mary
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing
      </p>
      <time
        className="reviews__time"
        dateTime="undefined"
      />
    </div>
  </li>
</ul>
`;
